### DISCUSSION TITLE
Exception "Application called an interface that was marshalled for a different thread" on Main program entry point

### DISCUSSION THREAD

**Discussion by AlbrechtWM** [dotnet/maui#7445] [Q&A] [answered]
I am getting an exception thrown here:

            global::Microsoft.UI.Xaml.Application.Start((p) => {
                var context = new global::Microsoft.UI.Dispatching.DispatcherQueueSynchronizationContext(global::Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread());
                global::System.Threading.SynchronizationContext.SetSynchronizationContext(context);
                new App();
            });

Before any of my user code is even being hit.  I don't know how this could be on the wrong thread since I assume this is the main thread on the main entry point of the program.  Any ideas what's going on?  See attached screenshot.

Thanks
<img width="1021" alt="maui issue" src="https://user-images.githubusercontent.com/3269909/170088937-22071ea2-2704-43eb-824d-794793651d9e.png">
.

**Comment by inforithmics** [comment, marked as answer]
I have the same issue

**Reply by inforithmics** [reply]
I solved by setting the Entry point: to `$targetentrypoint$`

<img width="198" alt="image" src="https://user-images.githubusercontent.com/2290863/171631278-635b6b81-df13-4103-882c-1857636666db.png">

<img width="65" alt="image" src="https://user-images.githubusercontent.com/2290863/171631356-c0ab321d-7f7e-4f84-9168-fc64560b5aeb.png">

**Reply by AlbrechtWM** [reply]
That worked!  Thanks for letting me know!
